### ISSUE TITLE
Enhance right-click on tray icon

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#149]
# Description
The tray icon currently has two options in the right-click menu: Toggle Window and Quit

Proposal is to enhance this with a couple of new entries:

* Show currently loaded databases
  * Indicate locked/unlocked status
  * When database is clicked open the window and focus on that tab
* Lock all databases
* Password Generator
  * Open the window and open the password generator widget
* Perform Auto-Type
  * Shown only if auto-type is compiled in
  * Performs action on last selected window similar to (CTRL+V) shortcut

# Mockup

![tray_menu_mockup](https://cloud.githubusercontent.com/assets/2809491/21777979/65dbe50a-d66f-11e6-97ff-563762268b88.png)

**Comment by Throne3d** [comment]
I feel like the "Loaded Databases" line is going to be weird – at least on Windows, I don't think I usually see entries in context menus used solely to group the following lines. It should be clear enough if they're databases with lock/unlock symbols next to them.

The database names should probably be truncated if they get too long, and possibly the list should be abbreviated if there are more than a certain number of databases open. I doubt it'll occur particularly often, but context menus getting unnecessarily large can be annoying and ugly.

Otherwise I think it looks good.

**Comment by droidmonkey** [comment]
Good feedback thanks

**Comment by CitDub** [comment]
Is possible to add support in this tray menu for TOTP similar to the "[tray TOTP plugin](https://github.com/victor-rds/KeeTrayTOTP)"?

Thanks.

**Comment by TheZ3ro** [comment]
@CitDub there is already an opened Issue about totp, this is not the right one.

**Comment by phoerious** [comment]
Yes, please open a separate issue. A TOTP generator is not generally out of scope of this application, but please create a dedicated feature request.

**Comment by droidmonkey** [comment]
If related to this tray menu upgrade please describe that part here

**Comment by TheZ3ro** [comment]
There is already a TOTP issue, I think what @CitDub means is using the tray menu to display TOTP token

**Comment by CitDub** [comment]
Yes @TheZ3ro , the right-click menu displays the accounts with a token, and when you click on it, the token is copied to the clipboard.

**Comment by dschrempf** [comment]
I would also like to have an option to unlock the database(s) in case they are locked (or a global hotkey to perform locking / unlocking).

**Comment by jeyca** [comment]
Is it possible to use the tray icon just for show/hide the window without context menu?

**Comment by droidmonkey** [comment]
Not right now, no, but I plan to add that with this implementation.

**Comment by Chealer** [comment]
@dschrempf wrote:
> I would also like to have an option to unlock the database(s) in case they are locked (or a global hotkey to perform locking / unlocking).

I don't see value in unlocking from the tray menu, but ticket #12463 requests a global locking shortcut.

**Comment by Chealer** [comment]
2.7.12 features the _Lock All Databases_ entry, although the shortcut does not work in Windows 11. I recommended to add a mnemonic instead in #12463.

As for _Perform Auto-Type_, that would be good, but it could be a more complete menu as offered in the window.

<details>
  <summary>License</summary>
All comments and attachments from myself on this ticket are offered under the terms of CC0 1.0.
</details>
